packages/react-select/src/Select.js

```diff
@@ -374,7 +375,18 @@ function buildCategorizedOptions(
     })
     // Flow limitation (see https://github.com/facebook/flow/issues/1414)
     .filter(categorizedOption => !!categorizedOption): any[]);
-}
+}, function([newProps, newSelectValue], [lastProps, lastSelectValue]) {
+  // Shallow compare props
+  // ignore menuIsOpen, as it's not used by sub-functions
+  // use find instead of filter(!"menuIsOpen").map(equal).includes(false)
+  // to traverse the properties as few times as possible
+  let propsEqual = !Object.entries(newProps).find(([key, value]) => key !== 'menuIsOpen' && value !== lastProps[key]);
```

Comment:
Optimally I would only compare the props that are needed by this function, but the whole prop object is passed to sub-functions, so it would be to hard to keep track of.

I don't check `menuIsOpen` because it's not used here and is changing when I need memoization of this function.